<!DOCTYPE html><html lang="en" ><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="Jekyll v4.0.0" /><meta property="og:title" content="Clinical decision support algorithm based on machine learning to assess the clinical response to anti–pd-1 therapy" /><meta property="og:locale" content="en_US" /><meta name="description" content="A tissue origin prediction device, method of predicting the tissue origin using a genome data, and computer program 1020200076756 · Filed Jun 23, 2020" /><meta property="og:description" content="A tissue origin prediction device, method of predicting the tissue origin using a genome data, and computer program 1020200076756 · Filed Jun 23, 2020" /><link rel="canonical" href="http://localhost:4000/2022/06/01/clinical-decision-support-algorithm-to-anti-pd-1-therapy" /><meta property="og:url" content="http://localhost:4000/2022/06/01/clinical-decision-support-algorithm-to-anti-pd-1-therapy" /><meta property="og:site_name" content="Sidey" /><meta property="og:type" content="article" /><meta property="article:published_time" content="2022-06-01T00:00:00+09:00" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Clinical decision support algorithm based on machine learning to assess the clinical response to anti–pd-1 therapy" /><meta name="twitter:site" content="@" /> <script type="application/ld+json"> {"@type":"BlogPosting","headline":"Clinical decision support algorithm based on machine learning to assess the clinical response to anti–pd-1 therapy","url":"http://localhost:4000/2022/06/01/clinical-decision-support-algorithm-to-anti-pd-1-therapy","dateModified":"2022-06-01T00:00:00+09:00","datePublished":"2022-06-01T00:00:00+09:00","mainEntityOfPage":{"@type":"WebPage","@id":"http://localhost:4000/2022/06/01/clinical-decision-support-algorithm-to-anti-pd-1-therapy"},"description":"A tissue origin prediction device, method of predicting the tissue origin using a genome data, and computer program 1020200076756 · Filed Jun 23, 2020","@context":"https://schema.org"}</script><title> Clinical decision support algorithm based on machine learning to assess the clinical response to anti–pd-1 therapy - Sidey</title><link rel="shortcut icon" href="/favicon.png"><link rel="alternate" type="application/atom+xml" title="Sidey" href="/atom.xml"><link rel="alternate" type="application/json" title="Sidey" href="http://localhost:4000/feed.json" /><link rel="sitemap" type="application/xml" title="sitemap" href="/sitemap.xml" /><style> *,:after,:before{box-sizing:border-box;background-color:inherit;color:inherit;margin:0;padding:0}body{font-family:system-ui, sans-serif;-webkit-font-smoothing:antialiased;text-rendering:optimizeLegibility;line-height:1.5;font-size:1rem;color:#16171a}nav ul{border-right:1px solid #edf2f7}a{color:#000;text-decoration-skip-ink:auto;text-decoration:underline}pre{margin:.5rem 0;padding:.5rem}.post p{margin:.5rem 0}.post h1,.post h2,.post h3,.post h4{margin:1rem 0}.post h2:first-child,.project h2:first-child,.photo h2:first-child{margin-top:0}.meta{margin:2rem 0}code,pre{background:#ecedee}code{padding:.1rem}pre code{border:none}pre{padding:1rem;overflow-x:auto}img{max-width:100%}hr{background:#000;height:1px;border:0}header{flex-basis:10rem;flex-grow:1;position:relative}header a{text-decoration:none}header li{margin-bottom:.2rem;text-align:right;margin-right:2rem}header a.active{font-weight:bold}header,section{padding:1rem}blockquote{font-style:italic;border-left:5px solid #ececec;padding-left:1rem}h1,h2,h3,h4,h5{line-height:1;margin:1rem 0;font-weight:600}section h1:first-child{margin-top:0}strong,b{font-weight:bold}.photos ul{list-style:none}.photos li{margin-bottom:1.5rem}.photo picture,.project picture{margin-bottom:0.5rem}.posts ul,header ul{list-style:none}.posts li{align-items:center;display:flex;justify-content:space-between;margin-bottom:.5rem}.posts li a,.posts li div,.projects li a{white-space:nowrap;overflow:hidden;text-overflow:ellipsis;text-decoration:none}.posts li time,.projects li time{padding-left:1rem;white-space:nowrap;font-variant-numeric:tabular-nums}main{display:flex;flex-wrap:wrap;max-width:60rem;margin:2rem auto;padding:1rem}@media screen and (max-width: 45rem){header li{display:inline;margin-right:1rem}.logo{padding-bottom:1rem}header ul{border-bottom:1px solid #edf2f7;padding-bottom:2rem}nav ul{border-right:0px}.photos ul{margin-top:0.5rem}}section{flex-basis:0;flex-grow:999;min-width:70%;display:flex;flex-direction:column}figcaption{font-size:smaller}</style></head><body><main role="main"><header role="banner"> <!--<h1 class="logo">Sidey</h1>--><nav role="navigation"><ul><li><a href="/" >Writing</a></li><li><a href="/about" >About</a></li><li><a href="/search" >Search</a></li><li><a href="/atom.xml" >Rss</a></li></ul></nav></header><section class="post"><h2>Clinical decision support algorithm based on machine learning to assess the clinical response to anti–pd-1 therapy</h2><h2 id="a-tissue-origin-prediction-device-method-of-predicting-the-tissue-origin-using-a-genome-data-and-computer-program">A tissue origin prediction device, method of predicting the tissue origin using a genome data, and computer program</h2><h3 id="1020200076756--filed-jun-23-2020">1020200076756 · Filed Jun 23, 2020</h3><h1 id="summary">Summary</h1><hr /><p>Anti-programmed death (PD)-1 therapy (αPD-1) has been used in patients with non-small cell lung cancer (NSCLC), leading to improved outcomes. However, the outcomes of therapy are still insufficient, and the expression of PD-ligand 1 (PD-L1) is not always a predictor of response to αPD-1.</p><p>The immuno-cancer drugs used for treatment purposes of NSCLC are typically Kitruda and Optivo. <br /> Each monthly treatment costs about $600,000 and $800,000, respectively, and costs nearly $10 million a year.</p><p>But the drug doesn’t respond to all patients. Therefore, in the case of Kitruda, the ‘PD-L1 expression positive rate’ must be 50% or more to be prescribed. However, even if the conditions are satisfied and prescribed, the actual drug reaction is about 60%.</p><p>I studied a drug activity prediction model using Non-FDA-approved information to help more patients get the correct prescription and reduce the cost of drug waste.</p><h1 id="summary-idea--solution">Summary Idea &amp; Solution</h1><hr /><h2 id="data">Data</h2><ul><li>Clinical data including patient characteristics</li><li>Mutations</li><li>Laboratory findings from the electronic medical records</li></ul><h2 id="data-feature">Data Feature</h2><ul><li>Missing values ​​exist in most features. Learning inputation models separately based on the distribution of other features and their non-missing values, rather than simple mean, zero, etc. methods.</li><li>To prevent data leakage, it was reviewed directly by the clinician.</li></ul><h2 id="model-selection--evaluation">Model Selection &amp; Evaluation</h2><ul><li>Proceed with model selection, including linear models, to find the appropriate complexity of the model.</li><li>Due to the small number of samples, LOOCV is used for model selection</li><li>Due to the nature of the field, some explanatory power is required, so the ensemble stage to maximize performance is not performed.</li><li>Finding samples that appear to be somewhat outliers as a result of LOOCV. (4 samples)</li></ul><h2 id="feature-attribution">Feature Attribution</h2><ul><li>It is very likely that there is an interaction effect between data.</li><li>Therefore, feature importance based on simple entropy is highly likely to cause errors in interpretation.</li><li>The Lime-based explanation model has a small number of samples, so it is judged that there is a lot of room for problems in fitting the local model.</li><li>For each sample, the average of the shap values ​​and the interaction value were calculated.</li></ul><p><img src="/assets/publication_patents/CDSS_main.jpg" alt="main_fig" /> <img src="/assets/publication_patents/paper_front.png" alt="front" /></p><h1 id="detail">Detail</h1><hr /><h2 id="feature-engineering">Feature engineering</h2><h3 id="gene-metastasis-feature">Gene&amp; Metastasis Feature</h3><p><strong>Sparsity</strong> <br /></p><p>When judging based on the commonly used sparsity judgment criteria, it was determined that there was sparsity by the ratio of zero value.</p><ul><li>Percentage of zero values: About $Gene Expression \in {0,1}$, The ratio of zero is at least 85% to a maximum of 95%. Therefore, it is judged to be sparse.</li><li>Number of observations: It does not see the sparse by observices.</li><li>Data Volatility: Because it is a binary, the conclusion is the same as that of Percenrage of zero. it is sparse.</li><li>Model performance: Unable to judge due to the limit of the number of samples.</li><li>Domain knowledge: Since there are no accurate statistics on gene expression for Korean, sparsity cannot be determined based on Domain knowledge.</li></ul><p><strong>New Feature</strong></p><ul><li>To reduce sparsity, change the gene feature to a new feature that represents the total number of expressions of the associated gene. \(\text{driver oncogene} = \sum{ \text{gene expression}}\)</li><li>Change Metastasis to “Total Metastasis” in the same way. \(\text{metastasis count} = \sum{ \text{metastasis present}}\)</li><li>Neutrophil and Lymphocyte were replaced by the LNR (lymphocyte-to-neutrophil ratio) feature. \(LNR = log(\frac{Neutrophil}{Lymphocyte} + \epsilon )\)</li></ul><p><br /></p><h2 id="model-selection">Model Selection</h2><p>Below is a comparison table of the eight models of the roc-auc score. It was evaluated with LOOCV (Leave one out CV). It was judged that the model required for prediction did not have to be highly complex. Therefore, we did not do model ensembles and compared only single models.<br /> <img src="/assets/publication_patents/paper_compare.png" alt="score" /> <br /></p><p><strong>Best Model Performance : LightGBM</strong></p><p><img src="/assets/publication_patents/paper_score.png" alt="score" /></p><p><br /></p><h2 id="feature-attribution-1">Feature Attribution</h2><hr /><p>It is the average value of the SHAP value of all samples.<br /> Shap is the contribution to the predicted value of a model considering the synergistic effect of one feature and another feature, from the point of view of game theory. <br /> Since the data in the medical and bio fields are features that are difficult to assert independence, it was judged that SHAP value-based interpretation was appropriate.</p><p><img src="/assets/publication_patents/paper_shap_val.png" alt="shap" /></p><p>Looking at the results, I suspected that the persistence of non-measurable lesions might have a high correlation with the target value. Thus, the persistence of non-measurable lesions is an ordinary type of data, so we look at spearman correlation. correlation = -0.44 and p-value = 8.1e-11. That is, it was not a strong relationship, so it was used as a feature.</p><span class="meta"><time datetime="2022-06-01T00:00:00+09:00">June 1, 2022</time> &middot; <a href="/tag/test">test</a></span></section></main></body></html>
